==59826== Memcheck, a memory error detector
==59826== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==59826== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==59826== Command: ./archivio 2 4
==59826== Parent PID: 59825
==59826== 
==59826== Warning: invalid file descriptor -1 in syscall read()
==59826== Warning: invalid file descriptor -1 in syscall read()
==59826== Thread 2:
==59826== Invalid read of size 4
==59826==    at 0x48F86E4: pthread_mutex_lock@@GLIBC_2.2.5 (pthread_mutex_lock.c:80)
==59826==    by 0x109D2D: tbody_signals_handler (archivio.c:207)
==59826==    by 0x48F5401: start_thread (pthread_create.c:442)
==59826==    by 0x4983743: clone (clone.S:100)
==59826==  Address 0x6400000029 is not stack'd, malloc'd or (recently) free'd
==59826== 
==59826== 
==59826== Process terminating with default action of signal 11 (SIGSEGV)
==59826==  Access not within mapped region at address 0x6400000029
==59826==    at 0x48F86E4: pthread_mutex_lock@@GLIBC_2.2.5 (pthread_mutex_lock.c:80)
==59826==    by 0x109D2D: tbody_signals_handler (archivio.c:207)
==59826==    by 0x48F5401: start_thread (pthread_create.c:442)
==59826==    by 0x4983743: clone (clone.S:100)
==59826==  If you believe this happened as a result of a stack
==59826==  overflow in your program's main thread (unlikely but
==59826==  possible), you can try to increase the size of the
==59826==  main thread stack using the --main-stacksize= flag.
==59826==  The main thread stack size used in this run was 8388608.
==59826== 
==59826== HEAP SUMMARY:
==59826==     in use at exit: 24,008,870 bytes in 35 blocks
==59826==   total heap usage: 36 allocs, 1 frees, 24,008,878 bytes allocated
==59826== 
==59826== Thread 1:
==59826== 16 bytes in 1 blocks are definitely lost in loss record 1 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10A251: main (archivio.c:52)
==59826== 
==59826== 24 bytes in 1 blocks are still reachable in loss record 2 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10BB8F: thread_create (thread.c:6)
==59826==    by 0x10A26E: main (archivio.c:57)
==59826== 
==59826== 24 bytes in 1 blocks are still reachable in loss record 3 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10BB8F: thread_create (thread.c:6)
==59826==    by 0x10A3B0: main (archivio.c:91)
==59826== 
==59826== 24 bytes in 1 blocks are still reachable in loss record 4 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10BB8F: thread_create (thread.c:6)
==59826==    by 0x10A73C: main (archivio.c:99)
==59826== 
==59826== 36 bytes in 1 blocks are still reachable in loss record 5 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x4026D8E: malloc (rtld-malloc.h:56)
==59826==    by 0x4026D8E: strdup (strdup.c:42)
==59826==    by 0x40144FD: _dl_load_cache_lookup (dl-cache.c:524)
==59826==    by 0x4008F99: _dl_map_object (dl-load.c:2142)
==59826==    by 0x400C928: dl_open_worker_begin (dl-open.c:534)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x400BE39: dl_open_worker (dl-open.c:782)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x400C20B: _dl_open (dl-open.c:886)
==59826==    by 0x49D2AF0: do_dlopen (dl-libc.c:95)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x49D2852: _dl_catch_error (dl-error-skeleton.c:227)
==59826== 
==59826== 36 bytes in 1 blocks are still reachable in loss record 6 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x400BB98: malloc (rtld-malloc.h:56)
==59826==    by 0x400BB98: _dl_new_object (dl-object.c:199)
==59826==    by 0x4006F9E: _dl_map_object_from_fd (dl-load.c:1063)
==59826==    by 0x4008A3F: _dl_map_object (dl-load.c:2275)
==59826==    by 0x400C928: dl_open_worker_begin (dl-open.c:534)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x400BE39: dl_open_worker (dl-open.c:782)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x400C20B: _dl_open (dl-open.c:886)
==59826==    by 0x49D2AF0: do_dlopen (dl-libc.c:95)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x49D2852: _dl_catch_error (dl-error-skeleton.c:227)
==59826== 
==59826== 48 bytes in 2 blocks are still reachable in loss record 7 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10BB8F: thread_create (thread.c:6)
==59826==    by 0x10A721: main (archivio.c:95)
==59826== 
==59826== 56 bytes in 1 blocks are still reachable in loss record 8 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x4002C64: malloc (rtld-malloc.h:56)
==59826==    by 0x4002C64: _dl_map_object_deps (dl-deps.c:463)
==59826==    by 0x400C98E: dl_open_worker_begin (dl-open.c:592)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x400BE39: dl_open_worker (dl-open.c:782)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x400C20B: _dl_open (dl-open.c:886)
==59826==    by 0x49D2AF0: do_dlopen (dl-libc.c:95)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x49D2852: _dl_catch_error (dl-error-skeleton.c:227)
==59826==    by 0x49D2C49: dlerror_run (dl-libc.c:45)
==59826==    by 0x49D2C49: __libc_dlopen_mode (dl-libc.c:162)
==59826==    by 0x4983350: __libc_unwind_link_get (unwind-link.c:50)
==59826== 
==59826== 64 bytes in 1 blocks are still reachable in loss record 9 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10A9A2: hash_table_create (hash_table.c:12)
==59826==    by 0x10A23D: main (archivio.c:49)
==59826== 
==59826== 80 bytes in 1 blocks are still reachable in loss record 10 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10B095: buffer_create (buffer.c:11)
==59826==    by 0x10A2DA: main (archivio.c:66)
==59826== 
==59826== 80 bytes in 1 blocks are still reachable in loss record 11 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10B095: buffer_create (buffer.c:11)
==59826==    by 0x10A2EB: main (archivio.c:67)
==59826== 
==59826== 80 bytes in 1 blocks are still reachable in loss record 12 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10A2FC: main (archivio.c:70)
==59826== 
==59826== 80 bytes in 1 blocks are still reachable in loss record 13 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10A344: main (archivio.c:79)
==59826== 
==59826== 96 bytes in 4 blocks are still reachable in loss record 14 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10BB8F: thread_create (thread.c:6)
==59826==    by 0x10A7C5: main (archivio.c:103)
==59826== 
==59826== 160 bytes in 1 blocks are still reachable in loss record 15 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10B07F: buffer_create (buffer.c:7)
==59826==    by 0x10A2DA: main (archivio.c:66)
==59826== 
==59826== 160 bytes in 1 blocks are still reachable in loss record 16 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10B07F: buffer_create (buffer.c:7)
==59826==    by 0x10A2EB: main (archivio.c:67)
==59826== 
==59826== 272 bytes in 1 blocks are possibly lost in loss record 17 of 27
==59826==    at 0x4849A83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x4012949: calloc (rtld-malloc.h:44)
==59826==    by 0x4012949: allocate_dtv (dl-tls.c:375)
==59826==    by 0x4012949: _dl_allocate_tls (dl-tls.c:634)
==59826==    by 0x48F5F99: allocate_stack (allocatestack.c:423)
==59826==    by 0x48F5F99: pthread_create@@GLIBC_2.34 (pthread_create.c:650)
==59826==    by 0x10BBB2: thread_create (thread.c:13)
==59826==    by 0x10A26E: main (archivio.c:57)
==59826== 
==59826== 272 bytes in 1 blocks are possibly lost in loss record 18 of 27
==59826==    at 0x4849A83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x4012949: calloc (rtld-malloc.h:44)
==59826==    by 0x4012949: allocate_dtv (dl-tls.c:375)
==59826==    by 0x4012949: _dl_allocate_tls (dl-tls.c:634)
==59826==    by 0x48F5F99: allocate_stack (allocatestack.c:423)
==59826==    by 0x48F5F99: pthread_create@@GLIBC_2.34 (pthread_create.c:650)
==59826==    by 0x10BBB2: thread_create (thread.c:13)
==59826==    by 0x10A3B0: main (archivio.c:91)
==59826== 
==59826== 272 bytes in 1 blocks are possibly lost in loss record 19 of 27
==59826==    at 0x4849A83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x4012949: calloc (rtld-malloc.h:44)
==59826==    by 0x4012949: allocate_dtv (dl-tls.c:375)
==59826==    by 0x4012949: _dl_allocate_tls (dl-tls.c:634)
==59826==    by 0x48F5F99: allocate_stack (allocatestack.c:423)
==59826==    by 0x48F5F99: pthread_create@@GLIBC_2.34 (pthread_create.c:650)
==59826==    by 0x10BBB2: thread_create (thread.c:13)
==59826==    by 0x10A73C: main (archivio.c:99)
==59826== 
==59826== 456 bytes in 1 blocks are still reachable in loss record 20 of 27
==59826==    at 0x4849A83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x4013CA3: calloc (rtld-malloc.h:44)
==59826==    by 0x4013CA3: _dl_check_map_versions (dl-version.c:280)
==59826==    by 0x400CB93: dl_open_worker_begin (dl-open.c:600)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x400BE39: dl_open_worker (dl-open.c:782)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x400C20B: _dl_open (dl-open.c:886)
==59826==    by 0x49D2AF0: do_dlopen (dl-libc.c:95)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x49D2852: _dl_catch_error (dl-error-skeleton.c:227)
==59826==    by 0x49D2C49: dlerror_run (dl-libc.c:45)
==59826==    by 0x49D2C49: __libc_dlopen_mode (dl-libc.c:162)
==59826==    by 0x4983350: __libc_unwind_link_get (unwind-link.c:50)
==59826== 
==59826== 472 bytes in 1 blocks are still reachable in loss record 21 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x48E01AD: __fopen_internal (iofopen.c:65)
==59826==    by 0x48E01AD: fopen@@GLIBC_2.2.5 (iofopen.c:86)
==59826==    by 0x10A2C0: main (archivio.c:62)
==59826== 
==59826== 544 bytes in 2 blocks are possibly lost in loss record 22 of 27
==59826==    at 0x4849A83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x4012949: calloc (rtld-malloc.h:44)
==59826==    by 0x4012949: allocate_dtv (dl-tls.c:375)
==59826==    by 0x4012949: _dl_allocate_tls (dl-tls.c:634)
==59826==    by 0x48F5F99: allocate_stack (allocatestack.c:423)
==59826==    by 0x48F5F99: pthread_create@@GLIBC_2.34 (pthread_create.c:650)
==59826==    by 0x10BBB2: thread_create (thread.c:13)
==59826==    by 0x10A721: main (archivio.c:95)
==59826== 
==59826== 800 bytes in 1 blocks are still reachable in loss record 23 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x10A9C8: hash_table_create (hash_table.c:20)
==59826==    by 0x10A23D: main (archivio.c:49)
==59826== 
==59826== 1,088 bytes in 4 blocks are possibly lost in loss record 24 of 27
==59826==    at 0x4849A83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x4012949: calloc (rtld-malloc.h:44)
==59826==    by 0x4012949: allocate_dtv (dl-tls.c:375)
==59826==    by 0x4012949: _dl_allocate_tls (dl-tls.c:634)
==59826==    by 0x48F5F99: allocate_stack (allocatestack.c:423)
==59826==    by 0x48F5F99: pthread_create@@GLIBC_2.34 (pthread_create.c:650)
==59826==    by 0x10BBB2: thread_create (thread.c:13)
==59826==    by 0x10A7C5: main (archivio.c:103)
==59826== 
==59826== 1,230 bytes in 1 blocks are still reachable in loss record 25 of 27
==59826==    at 0x4849A83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x400B89C: calloc (rtld-malloc.h:44)
==59826==    by 0x400B89C: _dl_new_object (dl-object.c:92)
==59826==    by 0x4006F9E: _dl_map_object_from_fd (dl-load.c:1063)
==59826==    by 0x4008A3F: _dl_map_object (dl-load.c:2275)
==59826==    by 0x400C928: dl_open_worker_begin (dl-open.c:534)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x400BE39: dl_open_worker (dl-open.c:782)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x400C20B: _dl_open (dl-open.c:886)
==59826==    by 0x49D2AF0: do_dlopen (dl-libc.c:95)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x49D2852: _dl_catch_error (dl-error-skeleton.c:227)
==59826== 
==59826== 2,304 bytes in 1 blocks are possibly lost in loss record 26 of 27
==59826==    at 0x4844899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x40046FB: malloc (rtld-malloc.h:56)
==59826==    by 0x40046FB: _dlfo_mappings_segment_allocate (dl-find_object.c:217)
==59826==    by 0x40046FB: _dl_find_object_update_1 (dl-find_object.c:671)
==59826==    by 0x40046FB: _dl_find_object_update (dl-find_object.c:805)
==59826==    by 0x400CB4F: dl_open_worker_begin (dl-open.c:735)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x400BE39: dl_open_worker (dl-open.c:782)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x400C20B: _dl_open (dl-open.c:886)
==59826==    by 0x49D2AF0: do_dlopen (dl-libc.c:95)
==59826==    by 0x49D2799: _dl_catch_exception (dl-error-skeleton.c:208)
==59826==    by 0x49D2852: _dl_catch_error (dl-error-skeleton.c:227)
==59826==    by 0x49D2C49: dlerror_run (dl-libc.c:45)
==59826==    by 0x49D2C49: __libc_dlopen_mode (dl-libc.c:162)
==59826==    by 0x4983350: __libc_unwind_link_get (unwind-link.c:50)
==59826== 
==59826== 24,000,096 bytes in 1 blocks are still reachable in loss record 27 of 27
==59826==    at 0x4849A83: calloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==59826==    by 0x497CBF7: hcreate_r (hsearch_r.c:97)
==59826==    by 0x10A992: hash_table_create (hash_table.c:8)
==59826==    by 0x10A23D: main (archivio.c:49)
==59826== 
==59826== LEAK SUMMARY:
==59826==    definitely lost: 16 bytes in 1 blocks
==59826==    indirectly lost: 0 bytes in 0 blocks
==59826==      possibly lost: 4,752 bytes in 10 blocks
==59826==    still reachable: 24,004,102 bytes in 24 blocks
==59826==         suppressed: 0 bytes in 0 blocks
==59826== 
==59826== For lists of detected and suppressed errors, rerun with: -s
==59826== ERROR SUMMARY: 8 errors from 8 contexts (suppressed: 0 from 0)
